mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/performance/MongoDBPerformanceTest.java

```diff
@@ -29,37 +29,40 @@
 @SkipByDatastoreProvider({ DatastoreProviderType.FONGO })
 public class MongoDBPerformanceTest extends OgmTestCase {
 
+	private static final String MONGO_COLLECTION = "com.mongodb.client.MongoCollection";
+	private static final String HELPER = "org.hibernate.ogm.utils.BytemanHelper";
+	private static final String BSON_DOCUMENT = "org.bson.conversions.Bson";
+
 	@Rule
 	public BytemanHelperStateCleanup bytemanState = new BytemanHelperStateCleanup();
 
 	@Test
 	@BMRules(rules = {
 			@BMRule(
-					targetClass = "com.mongodb.DBCollection",
-					targetMethod = "insert(java.util.List, com.mongodb.WriteConcern)",
-					helper = "org.hibernate.ogm.utils.BytemanHelper",
-					action = "countInvocation(\"update\")",
-					name = "countInsert"),
-			@BMRule(targetClass = "com.mongodb.DBCollection",
-					targetMethod = "update(com.mongodb.DBObject, com.mongodb.DBObject, boolean, boolean, com.mongodb.WriteConcern)",
-					helper = "org.hibernate.ogm.utils.BytemanHelper",
+					targetClass = MONGO_COLLECTION,
+					isInterface = true,
+					targetMethod = "insertMany(java.util.List)",
+					helper = HELPER,
 					action = "countInvocation(\"update\")",
-					name = "countUpdate"),
-			@BMRule(targetClass = "com.mongodb.DBCollection",
-					targetMethod = "update(com.mongodb.DBObject, com.mongodb.DBObject, boolean, boolean, com.mongodb.WriteConcern)",
-					helper = "org.hibernate.ogm.utils.BytemanHelper",
+					name = "countInsertMany"),
+			@BMRule(targetClass = MONGO_COLLECTION,
+					isInterface = true,
+					targetMethod = "updateOne(" + BSON_DOCUMENT + "," + BSON_DOCUMENT + ",com.mongodb.client.model.UpdateOptions)",
+					helper = HELPER,
 					action = "countInvocation(\"update\")",
-					name = "countUpdate"),
-			@BMRule(targetClass = "com.mongodb.DBCollection",
-					targetMethod = "findAndModify(com.mongodb.DBObject, com.mongodb.DBObject)",
-					helper = "org.hibernate.ogm.utils.BytemanHelper",
-					action = "countInvocation(\"update\")",
-					name = "countFindAndModify"),
-			@BMRule(targetClass = "com.mongodb.DBCollection",
-					targetMethod = "findOne(com.mongodb.DBObject, com.mongodb.DBObject, com.mongodb.ReadPreference)",
-					helper = "org.hibernate.ogm.utils.BytemanHelper",
+					name = "countUpdateOne"),
+			@BMRule(targetClass = MONGO_COLLECTION,
+					isInterface = true,
+					targetMethod = "find(" + BSON_DOCUMENT + ")",
+					helper = HELPER,
+					action = "countInvocation(\"load\")",
+					name = "countFind1"),
+			@BMRule(targetClass = MONGO_COLLECTION,
+					isInterface = true,
+					targetMethod = "find()",
+					helper = HELPER,
 					action = "countInvocation(\"load\")",
-					name = "countFindOne")
+					name = "countFind2")
 	})
```

Comment:
@DavideD  ... You see list of actual rules.  About name of the rule ....May be  You right.